app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpCreation.kt

```diff
@@ -0,0 +1,147 @@
+package com.github.palFinderTeam.palfinder.meetups.activities
+
+import android.annotation.SuppressLint
+import android.app.AlertDialog
+import android.content.Intent
+import android.icu.text.SimpleDateFormat
+import android.icu.util.Calendar
+import android.os.Bundle
+import android.view.View
+import android.widget.TextView
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import com.github.palFinderTeam.palfinder.R
+import com.github.palFinderTeam.palfinder.meetups.MeetUp
+import com.github.palFinderTeam.palfinder.meetups.TempUser
+import com.github.palFinderTeam.palfinder.utils.Location
+import com.github.palFinderTeam.palfinder.utils.askTime
+import com.github.palFinderTeam.palfinder.utils.isDeltaBefore
+
+const val MEETUP_EDIT = "com.github.palFinderTeam.palFinder.meetup_view.MEETUP_EDIT"
+const val defaultTimeDelta = 1000 * 60 * 60
+
+@SuppressLint("SimpleDateFormat") // Apps Crash with the alternative to SimpleDateFormat
+class MeetUpCreation : AppCompatActivity() {
+    private val model: MeetUpCreationViewModel by viewModels()
+    private var dateFormat = SimpleDateFormat()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_meet_up_creation)
+
+        dateFormat = SimpleDateFormat(getString(R.string.date_long_format))
+
+        if (intent.hasExtra(MEETUP_EDIT)) {
+            val meetup = intent.getSerializableExtra(MEETUP_EDIT) as MeetUp
+            fillFields(meetup)
+        } else {
+            updateDateFields()
+        }
+    }
+
+    private fun fillFields(meetUp: MeetUp){
+        setTextView(R.id.et_EventName, meetUp.name)
+        setTextView(R.id.et_Description, meetUp.description)
+
+        if (meetUp.hasMaxCapacity){
+            setTextView(R.id.et_Capacity, meetUp.capacity.toString())
+        }
+
+        setStartDate(meetUp.startDate)
+        setEndDate(meetUp.endDate)
+        model.capacity = meetUp.capacity
+        model.hasMaxCapacity = meetUp.hasMaxCapacity
+    }
+
+    private fun setTextView(id: Int, value: String){
+        findViewById<TextView>(id).apply { this.text = value }
+    }
+
+    private fun updateDateFields(){
+        setTextView(R.id.tv_StartDate, dateFormat.format(model.startDate))
+        setTextView(R.id.tv_EndDate, dateFormat.format(model.endDate))
+    }
+
+    /**
+     * Set Start Date on Model and update UI
+     */
+    private fun setStartDate(date: Calendar){
+        model.startDate = date
+        checkDateIntegrity()
+        updateDateFields()
```

Comment:
This is typically where LiveData could be use. It does work here, but using liveData you can simply tell the viewModel to update it's value through a clear API, and then you UI elements would observe the viewModel through LiveData and update automatically. In a simple case like here, it's not much of a deal but it might be worth considering elsewhere.